parsec/core/gui/file_table.py

```diff
@@ -269,7 +271,13 @@ def show_context_menu(self, pos):
             action = menu.addAction(_("ACTION_FILE_MENU_SHOW_FILE_STATUS"))
             action.triggered.connect(self.show_status_clicked.emit)
             action = menu.addAction(_("ACTION_FILE_MENU_GET_FILE_LINK"))
-            action.triggered.connect(self.file_path_clicked.emit)
+
+            # Show the option to create a timestamped share link to the user is
+            # useless here because it has the same effect as creating a regular file link
+            if not self.has_timestamped_file:
+                action.triggered.connect(self.file_path_clicked.emit)
+                action = menu.addAction(_("ACTION_FILE_MENU_GET_FILE_LINK_TIMESTAMP"))
+            action.triggered.connect(self.file_path_timestamp_clicked.emit)
```

Comment:
I would prefer it this was a little more explicit (ie: more ifs), because I can easily see someone adding an action at the wrong place later.
```python
action = menu.addAction(_("ACTION_FILE_MENU_GET_FILE_LINK"))
action = menu.addAction("My New Action")
action.triggered.connect(self.on_my_new_action_triggered)
if not self.has_timestamped_file:
    action.triggered.connect(self.file_path_clicked.emit)
    action = menu.addAction(_("ACTION_FILE_MENU_GET_FILE_LINK_TIMESTAMP"))
action.triggered.connect(self.file_path_timestamp_clicked.emit)
```